pkg/webui/components/map/index.js

```diff
@@ -34,75 +33,62 @@ Leaflet.Icon.Default.mergeOptions({
   shadowUrl: require('leaflet/dist/images/marker-shadow.png'),
 })
 
-export default class Map extends React.Component {
+export default class LocationMap extends React.Component {
   static propTypes = {
-    // Id is a string used to give the map a unique ID.
-    id: PropTypes.string.isRequired,
     // LeafletConfig is an object which can contain any number of properties defined by the leaflet plugin and is used to overwrite the default configuration of leaflet.
     leafletConfig: PropTypes.shape({}),
     // Markers is an array of objects containing a specific properties
     markers: PropTypes.arrayOf(
       // Position is a object containing two properties latitude and longitude which are both numbers.
       PropTypes.shape({
-        position: PropTypes.objectOf(PropTypes.number),
+        position: PropTypes.shape({
+          longitude: PropTypes.number,
+          latitude: PropTypes.number,
+        }),
       }),
-    ).isRequired,
+    ),
+    onClick: PropTypes.func,
     // Widget is a boolean used to add a class name to the map container div for styling.
     widget: PropTypes.bool,
   }
 
   static defaultProps = {
     leafletConfig: {},
     widget: false,
-  }
-
-  getMapCenter(markers) {
-    // This will calculate zoom and map center long/lang based on all markers provided.
-    // Currently it just returns the first marker.
-    // TODO: action (https://github.com/TheThingsNetwork/lorawan-stack/issues/1241)
-    return markers[0]
-  }
-
-  createMap(config, id) {
-    this.map = Leaflet.map(id, {
-      ...config,
-    })
-  }
-
-  createMarkers(markers) {
-    markers.map(marker =>
-      Leaflet.marker([marker.position.latitude, marker.position.longitude]).addTo(this.map),
-    )
-  }
-
-  componentDidMount() {
-    const { id, markers } = this.props
-
-    const { position } = markers.length >= 1 ? this.getMapCenter(markers) : markers[0]
-
-    const config = {
-      layers: [
-        Leaflet.tileLayer('http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
-          attribution: '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors',
-        }),
-      ],
-      center: [position.latitude, position.longitude],
-      zoom: 11,
-      minZoom: 1,
-      ...this.props.leafletConfig,
-    }
-
-    this.createMap(config, id)
-    this.createMarkers(markers, id)
+    markers: [],
+    onClick: undefined,
   }
 
   render() {
-    const { id, widget } = this.props
-
-    return (
-      <div className={style.container}>
-        <div className={classnames(style.map, { [style.widget]: widget })} id={id} />
-      </div>
+    const { widget, markers, onClick, leafletConfig } = this.props
+    const position =
+      markers.length >= 1
+        ? [markers[0].position.latitude, markers[0].position.longitude]
+        : [52.7, 4.9]
+    const map = (
+      <Map
+        className={classnames(style.map, { [style.widget]: widget })}
+        center={position}
+        zoom={7}
```

Comment:
can you try saving the zoom value as well? i think it might solve the issue with resetting zoom value after setting a marker

Reply:
fixed by storing the local state of the variable. had to move it under `{...leafletConfig}` in component to avoid overwriting in case the zoom value is set in a parent component.